crates/m3/src/builder/constraint_system.rs

```diff
@@ -155,23 +155,8 @@ impl<F: TowerField> ConstraintSystem<F> {
 	pub fn build_witness<'cs, 'alloc, P: PackedField<Scalar = F>>(
```

Comment:
At this point I would consider removing this method, and just calling `WitnessIndex::new(&cs.tables, &allocator)` where `cs.build_witness(&allocator)` is being used now. If we want to keep build_witness around, its return type no longer needs to be a Result, and this section from the docstring should be updated/removed, since the statement is no longer passed as an argument:
> 	/// The statement includes information about the tables sizes, which this requires in order to
>	/// allocate the column data correctly. The created witness index needs to be populated before
>	/// proving.

Reply:
Yeah, that makes sense. I'll leave this method to make it easier for existing code to migrate (there are some external codebases), but mark it deprecated. Will remove the Result type though, that doesn't make sense anymore.